recursive-feature-elimination/script.whizzml

```diff
@@ -0,0 +1,248 @@
+  ;;   test-dataset-id: (dataset-id) id of the dataset used to
+  ;;                     evaluate the model
+  ;;   evaluation-metric: (string) Metric  used in the evaluation
+  ;; Output: (map) Map with selected field names, selected field ids, and 
+  ;;         iterations evaluation info (only if test-dataset-id is present)
+  (define (recursive-feature-elimination dataset-id
+                                         nfeatures
+                                         objective-id
+                                         test-dataset-id
+                                         evaluation-metric)
+    (let (do-evaluation (not (empty? test-dataset-id))
+          obj-id (get-objective dataset-id objective-id)
+          fields (resource-fields dataset-id)
+          input-ids (default-inputs dataset-id obj-id))
+      (loop (selected input-ids
+             evaluations []) 
+          ;; Loop until we obtain the desired number of features
+          (if (<= (count selected) nfeatures)
+              (response selected 
+                        (feature-names fields selected) 
+                        evaluations 
+                        do-evaluation)
+              (let (_ (log-info "------- New iteration -------")
+                    _ (log-info "Creating random forest with " 
+                                (count selected) 
+                                " fields")
+                    ;; Create random forest and obtain its importances
+                    rf-id (random-forest dataset-id obj-id 0 selected)
+                    imps ((fetch rf-id) "importance")
+		    ;; Obtain fields with importance 0
+		    ;; (they  doesnt appear in "importance")
+		    no-imp-fields (filter (lambda (n)
+					    (not (member? n (keys imps))))
+					  selected)
+                    ;; Obtaining minimum field importance
+                    removed-field (min-key (lambda (x) (imps x)) (keys imps))
+		    removed-fields (cons removed-field no-imp-fields)
+		    removed-fields-names (feature-names fields removed-fields)
+                    _ (log-info  "Removed-fields " 
+                                 removed-fields ": "
+                                 removed-fields-names)
+                    ;; Remove field from selected list
+                    selected (keys (dissoc imps removed-field))
+                    _ (log-info "There are " 
+                                (count selected) 
+                                " fields after filtering")
+                    ev (when do-evaluation 
+                           (fetch(evaluate-model rf-id test-dataset-id)))
+                    ev-result (when ev (ev ["result" "model" evaluation-metric]))
+                    ;; Iterations evaluation info
+                    iterations (when ev {"features" (count selected)
+					 "evaluation"  ev-result
+					 "last-removed" removed-fields-names})
+                    evaluations (when ev (append evaluations iterations)))
```

Comment:
this `let` is quite long, and it's kind of difficult to follow what's going on... any chance you could define some functions to make it shorter? or maybe it's this long because of the many logs: perhaps we don't need that many?